contracts/staking/test/params.ts

```diff
@@ -0,0 +1,164 @@
+            expect(event.epochDurationInSeconds).to.bignumber.eq(_params.epochDurationInSeconds);
+            expect(event.rewardDelegatedStakeWeight).to.bignumber.eq(_params.rewardDelegatedStakeWeight);
+            expect(event.minimumPoolStake).to.bignumber.eq(_params.minimumPoolStake);
+            expect(event.maximumMakersInPool).to.bignumber.eq(_params.maximumMakersInPool);
+            expect(event.cobbDouglasAlphaNumerator).to.bignumber.eq(_params.cobbDouglasAlphaNumerator);
+            expect(event.cobbDouglasAlphaDenomintor).to.bignumber.eq(_params.cobbDouglasAlphaDenomintor);
+            // Assert `getParams()`.
+            const actual = await testContract.getParams.callAsync();
+            expect(actual[0]).to.bignumber.eq(_params.epochDurationInSeconds);
+            expect(actual[1]).to.bignumber.eq(_params.rewardDelegatedStakeWeight);
+            expect(actual[2]).to.bignumber.eq(_params.minimumPoolStake);
+            expect(actual[3]).to.bignumber.eq(_params.maximumMakersInPool);
+            expect(actual[4]).to.bignumber.eq(_params.cobbDouglasAlphaNumerator);
+            expect(actual[5]).to.bignumber.eq(_params.cobbDouglasAlphaDenomintor);
+        }
+
+        it('throws if not called by owner', async () => {
+            const tx = setParamsAndAssertAsync({}, notOwnerAddress);
+            const expectedError = new OwnableRevertErrors.OnlyOwnerError(notOwnerAddress, ownerAddress);
+            return expect(tx).to.revertWith(expectedError);
+        });
+
+        it('works if called by owner', async () => {
+            return setParamsAndAssertAsync({});
+        });
+
+        describe('rewardDelegatedStakeWeight', () => {
+            it('throws when > PPM_100_PERCENT', async () => {
+                const params = {
+                    // tslint:disable-next-line restrict-plus-operands
+                    rewardDelegatedStakeWeight: constants.PPM_100_PERCENT + 1,
+                };
+                const tx = setParamsAndAssertAsync(params);
+                const expectedError = new StakingRevertErrors.InvalidParamValueError(
+                    StakingRevertErrors.InvalidParamValueErrorCode.InvalidRewardDelegatedStakeWeight,
+                );
+                return expect(tx).to.revertWith(expectedError);
+            });
+        });
+
+        describe('maximumMakersInPool', () => {
+            it('throws when == 0', async () => {
+                const params = {
+                    maximumMakersInPool: 0,
+                };
+                const tx = setParamsAndAssertAsync(params);
+                const expectedError = new StakingRevertErrors.InvalidParamValueError(
+                    StakingRevertErrors.InvalidParamValueErrorCode.InvalidMaximumMakersInPool,
+                );
+                return expect(tx).to.revertWith(expectedError);
+            });
+        });
+
+        describe('cobb-douglas alpha', () => {
```

Comment:
I feel like this would be a good candidate for `combinatorial testing`. If you agree, I think it would make sense to parameterize these tests.

Reply:
Hmm, I don't think it's worth it. No real logic is behind these functions. We already fuzz test the Cobb Douglas function in its tests.